Lokad.AzureEventStore.Cache/MappedCacheProvider.cs

```diff
@@ -137,7 +137,7 @@ void DeleteFile(string fileName)
             Directory.CreateDirectory(Path.Combine(_directory, stateName));
 
             var path = Path.Combine(_directory, stateName, filename);
-            return (filename, new FileStream(path, FileMode.Create));
+            return (path, new FileStream(path, FileMode.Create));
```

Comment:
The previous behavior of `OpenWriteLocal` was to return a filename, not a path. As a consequence:

 - the documentation of the function needs to be updated
 - the variable names must be changed (from `filename` to `path`) in the signature
 - all uses of the function must be investigated to see if they relied on a filename being returned, and updated accordingly, in particular for `OpenReadAsync`.
 
It might be easier to fix the call to File.OpenRead on line 168 instead of changing the semantics of `OpenWriteLocal`.

Reply:
I have updated the variable names and function documentation to make things clearer.

My first reaction was indeed to change the line 168.
But after further inspection, I believe that fixing the function instead makes the other usage of the function (`OpenReadAsync` indeed) correct.
In the current state of the code I believe that the copy from Azure fails as well as the upload a fresh cache.
Fixing OpenWriteLocal should restore both aspects.
The function being private, I don't think I have missed any other use ?